src/test/features/claim/routes/pay.ts

```diff
@@ -462,7 +464,7 @@ describe('Claim issue: post payment callback receiver', () => {
             payServiceMock.resolveRetrieve('Success')
             draftStoreServiceMock.resolveSave()
             claimStoreServiceMock.resolveRetrieveClaimByExternalIdTo404HttpCode('Claim not found by external id')
-            claimStoreServiceMock.rejectSaveClaimForUser()
+            claimStoreServiceMock.resolveRetriveUserRoles()
```

Comment:
That is not acceptable. Test case says `should return 500 and render error page when cannot save claim` and this is what should have been simulated.

Reply:
Also add 2 extra test cases for:
- failure in retrieving user roles
- failure in calling feature toggle API